subsys/net/ip/tcp2.c

```diff
@@ -464,7 +649,12 @@ static size_t tcp_data_get(struct tcp *conn, struct net_pkt *pkt)
 
 	if (len > 0) {
 		if (conn->context->recv_cb) {
-			struct net_pkt *up = net_pkt_clone(pkt, K_NO_WAIT);
+			struct net_pkt *up =
+				net_pkt_clone(pkt, TCP_PKT_ALLOC_TIMEOUT);
+
+			if (!up) {
+				return 0;
```

Comment:
Returning 0 here, should be perhaps somehow handled in the upper code, i.e. not sending an ACK in such case.

Reply:
Yes, we need to handle that properly. I will add checks.

Reply:
I was actually wondering why do we call clone here at all, seems pointless

Reply:
Alternatively, we can just ref() it, hovewer I am not sure how the ip module would react to it after skipping, etc. Actually, the proper way is to do pull(), i.e. trim the data instead of skip. So, perhaps clone is right here, however we should pull().